src/components/forms/DateRangePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,159 @@
+    const { maxDate: maxStartDate } = startDatePickerProps
+    const parsedMaxStartDate = maxStartDate && parseDateString(maxStartDate)
+    const parsedCurrentEndDate = endDateInternalValue && parseDateString(endDateInternalValue)
+
+    if (parsedCurrentEndDate && parsedMaxStartDate) {
+      if (parsedCurrentEndDate.getTime() < parsedMaxStartDate.getTime()) {
+        return formatDate(parsedCurrentEndDate)
+      } else {
+        return formatDate(parsedMaxStartDate)
+      }
+    } else {
+      return (parsedCurrentEndDate && formatDate(parsedCurrentEndDate)) 
+        || (parsedMaxStartDate && formatDate(parsedMaxStartDate)) 
+        || undefined
+    }
+  }
+
+  const getMinEndDate = (): string | undefined => {
+    const { minDate: minEndDate } = endDatePickerProps
+    const parsedMinEndDate = minEndDate && parseDateString(minEndDate)
+    const parsedCurrentStartDate = startDateInternalValue && parseDateString(startDateInternalValue)
+
+    if (parsedCurrentStartDate && parsedMinEndDate) {
+      if (parsedCurrentStartDate.getTime() > parsedMinEndDate.getTime()) {
+        return formatDate(parsedCurrentStartDate)
+      } else {
+        return formatDate(parsedMinEndDate)
+      }
+    } else {
+      return (parsedCurrentStartDate && formatDate(parsedCurrentStartDate)) 
+        || (parsedMinEndDate && formatDate(parsedMinEndDate)) 
+        || undefined
+    }
+  }
+
+  const getDatePickerOnChangeFn = (
+    originalOnChangeFn: ((val?: string) => void) | undefined,
+    setStateInternalValueFn: React.Dispatch<React.SetStateAction<string | undefined>>
+  ): (val?: string) => void => {
+    return (externallyFormattedValue?: string | undefined): void => {
+      const parsedValue = 
+        externallyFormattedValue && parseDateString(externallyFormattedValue, DEFAULT_EXTERNAL_DATE_FORMAT)
+      
+      if (parsedValue) {
+        // The externally input and formatted value is a valid date.
+        // Convert to internal format and set the internal state to
+        // the selected date.
+        const internallyFormattedValue = formatDate(parsedValue)
+        setStateInternalValueFn(internallyFormattedValue)
+      } else {
+        // Externally input and formatted value is not a valid date.
+        // Do not attempt to convert to internal date format. 
+        // Simply update internal state with the input value as received.
+        setStateInternalValueFn(externallyFormattedValue)
+      }
+
+      if (originalOnChangeFn) originalOnChangeFn(externallyFormattedValue)
+    }
+  }
+
+  const startDatePickerOnChange = getDatePickerOnChangeFn(
+    startDatePickerProps.onChange, 
+    setStartDateInternalValue
+  )
+
+  const endDatePickerOnChange = getDatePickerOnChangeFn(
+    endDatePickerProps.onChange,
+    setEndDateInternalValue
+  )
+  
+  const classes = classnames(className, 'usa-date-range-picker')
+
+  return (
+    <div className={classes} data-testid="date-range-picker">
+      <div className="usa-form-group">
```

Comment:
Could we use our own `FormGroup` & `Label` components here? I'm thinking of how we might want to expand this component in order to display an error state for example, and since those components already take an `error` prop that might make it easier

Reply:
Passed those classes through in bd99ba0, as well as updating to using our own FormGroup and Label components